create_ticket.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+
+import sys
+import argparse
+
+from common import get_envvar
+from jiralib import Jira
+from githublib import Github
+
+# max github pages to process
+GITHUB_MAX_PAGES = 20
+# process upto x weeks back
+max_delta_weeks = 4
+# how many tickets can be opened on each cycle
+flood_protection_limit = 3
+
+
+def process_issue(jira, github, issue, dryrun):
+    html_url = issue["html_url"]
+
+    if "pull" in html_url:
+        return False
+
+    if not github.check_labels(issue):
+        return False
+
+    issue_id = issue["number"]
+    if not jira.issue_exists(github.repo, issue_id):
+        if not github.check_time(issue, max_delta_weeks):
+            return False
+
+        if not dryrun:
+            created_issue = jira.create_issue(
+                github.repo, issue_id, issue["title"], html_url
+            )
+            issue_url = f"{jira.server}/browse/{created_issue}"
+            print(f"Created issue {issue_url} for {html_url}")
+        else:
+            print(f"Dry Run Created issue for {html_url}")
+        return True
+    else:
+        print("Issue for", html_url, "already exists")
+
+    return False
+
+
+def loop(jira, github, dryrun):
+    issues_created = 0
+
+    for page in range(1, GITHUB_MAX_PAGES):
+        issues = github.get_issues(page)
+        if len(issues) == 0:
+            break
+
+        for issue in issues:
+            res = process_issue(jira, github, issue, dryrun)
+            if res:
+                issues_created += 1
+                if issues_created == flood_protection_limit:
+                    print("Flood protection limit reached, exiting")
+                    sys.exit(0)
```

Comment:
you count only for issues that you create, there might be issues that are not sig/nework , or already created, so you don't count them

Reply:
Ok, so just do not mix the things in one.
Per what you are saying, we need to see:
- Fetch all issues.
- Filter issues per criteria.
- Open Jira issues per the filtered list.

If I understand correctly, the filtered list needs to be limited in size.

In python you can do all this even without wasting memory by using generators (for the outside, it looks like you process lists, but it actually iterates each time).

Reply:
I will refactor it, but note that we should not fetch all issues, because we might get to the threshold before we fetched them all, this is why we fetch by page, and break when the condition reached.
First version might still not use generators if its ok.